go/master/client.go

```diff
@@ -20,6 +25,51 @@ type Client struct {
 	ch   chan []byte
 }
 
+// MasterAddresser provide master address
+type masterAddresser struct {
+	client    *clientv3.Client
+	timeout   time.Duration
+	endpoints []string
+}
+
+// Address return the address
+func (m masterAddresser) Address() string {
+	for {
+		ctx, cancel := context.WithTimeout(context.Background(), m.timeout)
+		resp, err := m.client.Get(ctx, masterAddrPath)
```

Comment:
> Clients should use watch to retrieve notification about master node events
I think it's a good idea, it will reduce the count for `GET` operator to etcd.

> masterMonitor should also change to handle these events

I do not think `masterMonitor` will handle the events, we can handle these in a new method named `masterAddresser.init()`, and `masterAddresser.Addresser()` only return the latest master address.